src/Servers/Kestrel/Core/src/Internal/Http2/Http2OutputProducer.cs

```diff
@@ -425,10 +425,16 @@ private void ThrowIfSuffixSent()
         {
             if (_suffixSent)
             {
-                throw new InvalidOperationException("Writing is not allowed after writer was completed.");
+                ThrowSuffixSent();
             }
         }
 
+        [StackTraceHidden]
+        private static void ThrowSuffixSent()
```

Comment:
It's the same as in the `Http1OutputProducer`. Maybe reuse that method or move it to a `ThrowHelper`-class?